src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,167 @@
+import React, {
+  FC,
+  CSSProperties,
+  useState,
+  useCallback,
+  useRef,
+  ReactNode,
+  useEffect,
+} from 'react';
+import { createPortal } from 'react-dom';
+
+export interface DialogProps {
+  // the on/off switch of the component
+  open?: boolean;
+
+  // this attribute allows user to drag the dialog or not
+  draggable?: boolean;
+
+  // classes allows style overwritting
+  classes?: object | string;
+
+  // children must be ReactNode
+  children?: ReactNode;
+
+  // Callback function fired when the component requests to be closed.
+  onClose?: (val: any) => void;
+
+  // props for setting width of the modal
+  maxWidth?: string | boolean;
+
+  // props for setting full screen component
+  fullScreen?: boolean;
+
+  // custom styling
+  style?: CSSProperties;
+}
+
+/**
+ * Dialogs inform users about a task and can contain critical information, require decisions, or involve multiple tasks.
+ *
+ * ```js
+ * import { Dialog } from 'pat-ui'
+ * ```
+ */
+const Dialog: FC<DialogProps> = (props) => {
+  const {
+    open,
+    onClose,
+    children,
+    draggable,
+    classes,
+    maxWidth,
+    fullScreen,
+    style,
+  } = props;
+
+  // do logic in following lines
+
+  const [visible, setVisible] = useState<boolean>(open as boolean);
+  useEffect(() => {
+    setVisible(open as boolean);
+  }, [open]);
+  // use wrapper to get reference to the dialog container
+  const wrapperRef = useRef<any>(null);
+
+  // the logic that is used to let us pass value from Dialog via onClose()
+  const closeModal = useCallback(
+    ({ target }) => {
+      if (
+        wrapperRef &&
+        wrapperRef.current &&
+        !wrapperRef.current.contains(target) &&
+        onClose
+      ) {
+        onClose('Modal Closed');
+      }
+    },
+    [onClose]
+  );
+
+  // the logic that allows modal to be closed
+  useEffect(() => {
+    document.addEventListener('click', closeModal, { capture: true });
+    return () => {
+      document.removeEventListener('click', closeModal, { capture: true });
+    };
+  }, [closeModal]);
+
+  // logic that prevent the user from scrolling when clicked
+  useEffect(() => {
+    if (open) {
+      document.body.style.overflowY = 'hidden';
+    } else {
+      document.body.style.overflowY = '';
+    }
+  }, [open]);
+
+  const sizingWidth = (width: string | boolean) => {
+    switch (width) {
+      case false:
+        return 'auto';
+      case 'xs':
+        return '10%';
+      case 'sm':
+        return 'auto';
+      case 'md':
+        return '60%';
+      case 'lg':
+        return '80%';
+      case 'xl':
+        return '100%';
+      default:
+        return 'auto';
+    }
+  };
+  const widthOutput = sizingWidth(maxWidth as string | boolean);
+  const widthOption: React.CSSProperties = {
+    width: widthOutput,
+  };
+
+  // fullview logic
```

Comment:
you mean full screen here?

Reply:
yes MUI they call it full view as props name